sys/net/gnrc/netif/init_devs/auto_init_kw2xrf.c

```diff
@@ -41,20 +42,41 @@
 #define KW2XRF_NUM ARRAY_SIZE(kw2xrf_params)
 
 static kw2xrf_t kw2xrf_devs[KW2XRF_NUM];
+static netdev_ieee802154_submac_t kw2xrf_netdev[KW2XRF_NUM];
 static char _kw2xrf_stacks[KW2XRF_NUM][KW2XRF_MAC_STACKSIZE];
+gnrc_netif_bhp_ctx_t kw2xrf_ctx[KW2XRF_NUM];
 
-static gnrc_netif_t _netif[KW2XRF_NUM];
+static void kw2xrf_irq_event_handler(event_t *evt){
+    gnrc_netif_bhp_ctx_t *c = container_of(evt, gnrc_netif_bhp_ctx_t, event);
+    netdev_t *netdev = c->netif.dev;
+    netdev_ieee802154_t *netdev_ieee802154 = container_of(netdev, netdev_ieee802154_t, netdev);
+    netdev_ieee802154_submac_t *netdev_submac = container_of(netdev_ieee802154,
+                                                             netdev_ieee802154_submac_t,
+                                                             dev);
+    kw2xrf_radio_hal_irq_handler(&netdev_submac->submac.dev);
+}
 
 void auto_init_kw2xrf(void)
 {
     for (unsigned i = 0; i < KW2XRF_NUM; i++) {
         const kw2xrf_params_t *p = &kw2xrf_params[i];
 
         LOG_DEBUG("[auto_init_netif] initializing kw2xrf #%u\n", i);
-        kw2xrf_setup(&kw2xrf_devs[i], (kw2xrf_params_t*) p, i);
-        gnrc_netif_ieee802154_create(&_netif[i], _kw2xrf_stacks[i], KW2XRF_MAC_STACKSIZE,
+
+        /* Init radio */
+        kw2xrf_init(&kw2xrf_devs[i], (kw2xrf_params_t*) p,&kw2xrf_netdev[i].submac.dev,
+                        gnrc_netif_bhp_irq_handler, &kw2xrf_ctx[i]);
+
+        netdev_register(&kw2xrf_netdev[i].dev.netdev, NETDEV_KW2XRF, i);
+        netdev_ieee802154_submac_init(&kw2xrf_netdev[i]);
+
+        /* Setup events */
+        gnrc_netif_bhp_init(&kw2xrf_ctx[i], kw2xrf_irq_event_handler);
+
+        gnrc_netif_ieee802154_create(&kw2xrf_ctx[i].netif, _kw2xrf_stacks[i], KW2XRF_MAC_STACKSIZE,
                                      KW2XRF_MAC_PRIO, "kw2xrf",
-                                     &kw2xrf_devs[i].netdev.netdev);
+                                     &kw2xrf_netdev[i].dev.netdev);
+
     }
 }
 /** @} */
```

Comment:
Does LWIP also need an update?

Reply:
there's not yet support for this radio there :(. I can provide a port in a follow up

Reply:
Would be good to make sure the new bottom-half processing works with either stack

Reply:
Probably a middle layer could do the trick